app/services/anonymise_user.rb

```diff
@@ -0,0 +1,22 @@
+class AnonymiseUser
+  attr_accessor :user
+
+  def initialize(user:)
+    self.user = user
+  end
+
+  def call
+    result = Result.new(true)
+
+    User.transaction do
+      user.anonymised_at = DateTime.now
```

Comment:
Possibly a guard `raise` needed to protect in the circumstance that the user is not deactivated.